sdk/cosmos/azure-cosmos-spark_3-5_2-12/src/main/scala/com/azure/cosmos/spark/CosmosWriter.scala

```diff
@@ -0,0 +1,109 @@
+                            userConfig: Map[String, String],
+                            cosmosClientStateHandles: Broadcast[CosmosClientMetadataCachesSnapshots],
+                            diagnosticsConfig: DiagnosticsConfig,
+                            inputSchema: StructType,
+                            partitionId: Int,
+                            taskId: Long,
+                            epochId: Option[Long],
+                            sparkEnvironmentInfo: String)
+  extends CosmosWriterBase(
+    userConfig,
+    cosmosClientStateHandles,
+    diagnosticsConfig,
+    inputSchema,
+    partitionId,
+    taskId,
+    epochId,
+    sparkEnvironmentInfo
+  ) with OutputMetricsPublisherTrait {
+
+  private val recordsWritten = new AtomicLong(0)
+  private val bytesWritten = new AtomicLong(0)
+  private val totalRequestCharge = new AtomicLong(0)
+
+  private val  recordsWrittenMetric = new CustomTaskMetric {
+    override def name(): String = CosmosConstants.MetricNames.RecordsWritten
+    override def value(): Long = recordsWritten.get()
+  }
+
+  private val bytesWrittenMetric = new CustomTaskMetric {
+    override def name(): String = CosmosConstants.MetricNames.BytesWritten
+
+    override def value(): Long = bytesWritten.get()
+  }
+
+  private val totalRequestChargeMetric = new CustomTaskMetric {
+    override def name(): String = CosmosConstants.MetricNames.TotalRequestCharge
+
+    // Internally we capture RU/s up to 2 fractional digits to have more precise rounding
+    override def value(): Long = totalRequestCharge.get() / 100L
+  }
+
+  private val metrics = Array(recordsWrittenMetric, bytesWrittenMetric, totalRequestChargeMetric)
+
+  override def currentMetricsValues(): Array[CustomTaskMetric] = {
+    metrics
+  }
+
+  override def getOutputMetricsPublisher(): OutputMetricsPublisherTrait = this
+
+  override def trackWriteOperation(recordCount: Long, diagnostics: Option[CosmosDiagnosticsContext]): Unit = {
+    if (recordCount > 0) {
+      recordsWritten.addAndGet(recordCount)
+    }
+
+    diagnostics match {
+      case Some(ctx) =>
+        // Capturing RU/s with 2 fractional digits internally
+        totalRequestCharge.addAndGet((ctx.getTotalRequestCharge * 100L).toLong)
+        bytesWritten.addAndGet(
+          if (ImplementationBridgeHelpers
+            .CosmosDiagnosticsContextHelper
+            .getCosmosDiagnosticsContextAccessor
+            .getOperationType(ctx)
+            .isReadOnlyOperation) {
+
+            ctx.getMaxRequestPayloadSizeInBytes + ctx.getMaxResponsePayloadSizeInBytes
+          } else {
+            ctx.getMaxRequestPayloadSizeInBytes
+          }
+        )
+      case None =>
+    }
+  }
+
+  override def commit(): WriterCommitMessage = {
+    val commitMessage = super.commit()
+
+    // TODO @fabianm - this is a workaround - it shouldn't be necessary to do this here
+    // Unfortunately WriteToDataSourceV2Exec.scala is not updating custom metrics after the
```

Comment:
Again - no need to react to in this PR - but can you please double-check whether the issue mentioned here has been addressed in Spark 3.5 - and then we can react accordingly?

Reply:
The issue has not been addressed yet.